src/actions/editUserActions.js

```diff
@@ -0,0 +1,171 @@
+let { writeFile, readFile } = require("../helpers/helpers")
+
+//metoda folosita in reqValidData pentru a verifica daca mailul deja exista
+let checkMail = (newMail, allMails) => {
+    for (let i = 0; i < allMails.length; i++) {
+        if (newMail === allMails[i].email) return 1;
+    }
+    return 0;
+};
+
+//metoda verifica daca elementele introduse de user sunt corecte si corespund campului corespunzator
+let reqValidData = (str, putMethod,users) => {
+    //putMethod=1 daca aplicam metoda pe put
+    //putMethod=0 daca apliam metodaspe patch
+    let ok = "";
+   // let users = require("../../db/users.json");
+
+    if (str.firstName) {
+        if (!/^[a-zA-Z]+$/.test(str.firstName))
+            ok += "Ati introdus prenumele gresit" + "\n";
+    }
+    else if (putMethod) {
+        ok += "Nu ati introdus niciun prenume. \n"
+    }
+
+    if (str.lastName) {
+        if (!/^[a-zA-Z]+$/.test(str.lastName))
+            ok += "Ati introdus numele gresit" + "\n";
+    }
+    else if (putMethod) {
+        ok += "Nu ati introdus niciun nume. \n"
+    }
+
+    if (str.email) {
+        if (!/^([a-z0-9])+\@([a-z])+\.([a-z])+$/.test(str.email))
+            ok += "Ati introdus emailul gresit" + "\n";
+
+        if (checkMail(str.email, users) === 1) {
+            ok += "Mailul folosit exista deja in baza de date";
+        }
+    }
+    else if (putMethod) {
+        ok += "Nu ati introdus niciun email \n"
+    }
+
+    if (str.phone) {
+        if (!/^([0]{1})\d{5,9}$/.test(str.phone))
+            ok += "Ati introdus gresit numarul de telefon";
+    }
+    else if (putMethod) {
+        ok += "Nu ati introdus niciun numar de telefon \n"
+    }
+
+    return (ok === "") ? 1 : ok;
+};
+
+
+//returneaza linia unde se afla id-ul cautat 
```

Comment:
Comments usually indicate that the code itself is not self-descriptive. When you have the tendency to write comments thing about how would you name your function (or rewrite the code) to "say" what the comment says.
Ex: **getIdIndex** or **getIdIndexInArray**

Reply:
Am inlocuit cu getIdIndex